stablehlo/dialect/Base.h

```diff
@@ -255,6 +255,13 @@ void writeEnumAttribute(EnumTypeAttr val, DialectBytecodeWriter &writer) {
 }
 }  // namespace bytecode
 
+// Determines the speculatability for a shaped operation `op` with `shapeCount`
+// shape operands. The last `count` operands are assumed to be shape operands.
+// To be speculatable, such an op must either have a fully dynamic result type
```

Comment:
From the implementation it seems that whether the result is dynamic or not does not matter if `have only static inputs and constant shape operands.` is satisfied?